subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -17,20 +17,26 @@
 
 This chapter provides the information you need to migrate your older Gradle 4.x builds to Gradle 5.0. In most cases, you will need to apply the changes from all versions that come after the one you're upgrading from. For example, if you're upgrading from Gradle 4.3, you will also need to apply the changes since 4.4, 4.5, etc.
 
-First, a few pointers about upgrading:
+We recommend the following steps for all users:
 
- * Start by running `gradle --scan build` and viewing the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan, or by running `gradle --warn --warning-mode=all build` to see the deprecations in the console.
+ . If you are not already on version 4.10.2, follow the directions <<changes_5.0,below>> to upgrade your project to 4.10.2.
+ . Try running `gradle help --scan` and view the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan.
++
+image::https://gradle.com/wp-content/uploads/2018/08/deprecations-1.png[Deprecations View of a Gradle Build Scan]
 +
 This is so that you can see any deprecation warnings that apply to your build. Gradle 5.x will generate (potentially less obvious) errors if you try to upgrade directly to it. 
 +
-If your build doesn't have a `build` task, run whichever task or set of tasks does a full build.
- * Update your plugins.
+Alternatively, you could run `gradle help --warning-mode=all` to see the deprecations in the console, though it may not report as much detailed information.
+. Update your plugins.
 +
 Some plugins will break with this new version of Gradle, for example because they use internal APIs that have been removed or changed. The previous step will help you identify potential problems by issuing deprecation warnings when a plugin does try to use a deprecated part of the API.
 +
 In particular, you will need to use at least a 2.x version of the https://plugins.gradle.org/plugin/com.github.johnrengelman.shadow[*Shadow Plugin*].
+. Run `gradle wrapper --gradle-version 5.0` to update the project to 5.0
+. Move to Java 8 or higher if you haven't already. Whereas Gradle 4.x requires Java 7, Gradle 5 requires Java 8 to run.
+. Try to run the project and debug any errors.
 
```

Comment:
Worthwhile to link to the troubleshooting page here?

Reply:
done on the line above so it didn't close the comment.